test/test_SimulatedLinkamT95.py

```diff
@@ -0,0 +1,236 @@
+        self.assertNotEqual(status_bytes[6:10], '01b8')     # Temp != 44.0 C
+
+        # Finish off heating (and overshoot a bit)
+        linkam.process(5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[6:10], '01b8')    # Temp == 44.0 C
+
+        # Should hold now, so temperature should not change
+        linkam.process(10)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x30')       # Auto-holding at limit
+        self.assertEqual(status_bytes[6:10], '01b8')    # Temp == 44.0 C
+
+    def test_simple_cool(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to cool from 24.0 C to 4.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Cool for almost a minute but not quite
+        linkam.process(59.5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x20')           # Cooling status set
+        self.assertNotEqual(status_bytes[6:10], '0028')     # Temp != 4.0 C
+
+        # Finish off cooling (and overshoot a bit)
+        linkam.process(5)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[6:10], '0028')    # Temp == 4.0 C
+
+        # Should hold now, so temperature should not change
+        linkam.process(10)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x30')       # Auto-holding at limit
+        self.assertEqual(status_bytes[6:10], '0028')    # Temp == 4.0 C
+
+    def test_error_flag_overcool(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Ensure flag is not set
+        status_bytes = linkam.getStatus()
+        self.assertFalse(ord(status_bytes[1]) & 0x01)
+
+        # Set up to cool from 24.0 C to 4.0 C at 51.00 C/min
+        linkam.setRate('5100')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Ensure flag is set after running a bit
+        linkam.process(0.1)
+        status_bytes = linkam.getStatus()
+        self.assertTrue(ord(status_bytes[1]) & 0x01)
+
+    def test_stop_command(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to heat from 24.0 C to 44.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('440')
+        linkam.start()
+        linkam.process(0)
+
+        # Process for some time and then stop
+        linkam.process(10)
+        linkam.stop()
+        linkam.process(0)
```

Comment:
> It's not really a time based simulation. It's a cycle based simulation.

I fell into the trap :) You're totally right, it's cycle-based not time based. I got myself quite confused apparently. After that reminder I actually agree that there shouldn't be a separate function for this, not at all, and yes, maybe we can find a better name instead of `process`.

You're also right about `process` never being called anywhere except in unit tests, but there it might be called with "I don't care" often (as seen here). So what about making `dt` optional, possibly with `0` as default?

I think even though this was off-topic for the PR it was good that we cleared that up somehow, if anything I re-learned that it's based on cycles and not time :D